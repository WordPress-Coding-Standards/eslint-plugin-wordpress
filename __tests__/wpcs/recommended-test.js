'use strict';

const eslint = require( 'eslint' );
const path = require( 'path' );

const configFile = path.resolve( __dirname, '../../lib/configs/recommended.js' );
const linter = new eslint.CLIEngine({
	configFile,
	ignorePath: './.gitignore',
	useEslintrc: false,
});

describe( 'Recommended config tests', () => {
	let report;

	beforeEach(() => {
		report = linter.executeOnFiles(['__tests__/wpcs/fixtures/*-fixture.js']);
	});

	test( 'should have correct error and warning counts', () => {
		expect( report.results.length ).toBe( 2 );
		expect( report.errored ).toBeFalsy();
		expect( report.errorCount ).toBe( 8 );
		expect( report.warningCount ).toBe( 0 );
		expect( report.fixableErrorCount ).toBe( 6 );
		expect( report.fixableWarningCount ).toBe( 0 );
	});

	test( 'lint results should match snapshot', () => {
		let { results } = report;
		for (let result of results) {
			let relativeFilePath = path.relative( __dirname, result.filePath );
			expect( result ).toMatchSnapshot( relativeFilePath );
		}
	});
});
