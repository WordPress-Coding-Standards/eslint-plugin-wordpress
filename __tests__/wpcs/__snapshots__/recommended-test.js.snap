// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fixtures/semicolons-fixture.js 1`] = `
"error: Missing semicolon (semi) at __tests__/wpcs/fixtures/semicolons-fixture.js:3:7:
  1 | /* eslint-disable no-unused-vars */
  2 | var aa;
> 3 | var bb
    |       ^
  4 | var cc ;
  5 | var dd;;
  6 | var ee; ;


error: Unexpected whitespace before semicolon (semi-spacing) at __tests__/wpcs/fixtures/semicolons-fixture.js:4:8:
  2 | var aa;
  3 | var bb
> 4 | var cc ;
    |        ^
  5 | var dd;;
  6 | var ee; ;
  7 | /* eslint-enable no-unused-vars */


error: Missing whitespace after semicolon (semi-spacing) at __tests__/wpcs/fixtures/semicolons-fixture.js:5:7:
  3 | var bb
  4 | var cc ;
> 5 | var dd;;
    |       ^
  6 | var ee; ;
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Unnecessary semicolon (no-extra-semi) at __tests__/wpcs/fixtures/semicolons-fixture.js:5:8:
  3 | var bb
  4 | var cc ;
> 5 | var dd;;
    |        ^
  6 | var ee; ;
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Unnecessary semicolon (no-extra-semi) at __tests__/wpcs/fixtures/semicolons-fixture.js:6:9:
  4 | var cc ;
  5 | var dd;;
> 6 | var ee; ;
    |         ^
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Unnecessary escape character: \\\\\\" (no-useless-escape) at __tests__/wpcs/fixtures/strings-fixture.js:5:42:
  3 | var bb = 'Escape single quotes \\\\'within strings\\\\'';
  4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
> 5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
    |                                          ^
  6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Unnecessary escape character: \\\\\\" (no-useless-escape) at __tests__/wpcs/fixtures/strings-fixture.js:5:58:
  3 | var bb = 'Escape single quotes \\\\'within strings\\\\'';
  4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
> 5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
    |                                                          ^
  6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Strings must use singlequote (quotes) at __tests__/wpcs/fixtures/strings-fixture.js:6:10:
  4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
  5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
> 6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";
    |          ^
  7 | /* eslint-enable no-unused-vars */
  8 | 


8 errors found.
6 errors potentially fixable with the \`--fix\` option."
`;

exports[`fixtures/strings-fixture.js 1`] = `
"error: Missing semicolon (semi) at __tests__/wpcs/fixtures/semicolons-fixture.js:3:7:
  1 | /* eslint-disable no-unused-vars */
  2 | var aa;
> 3 | var bb
    |       ^
  4 | var cc ;
  5 | var dd;;
  6 | var ee; ;


error: Unexpected whitespace before semicolon (semi-spacing) at __tests__/wpcs/fixtures/semicolons-fixture.js:4:8:
  2 | var aa;
  3 | var bb
> 4 | var cc ;
    |        ^
  5 | var dd;;
  6 | var ee; ;
  7 | /* eslint-enable no-unused-vars */


error: Missing whitespace after semicolon (semi-spacing) at __tests__/wpcs/fixtures/semicolons-fixture.js:5:7:
  3 | var bb
  4 | var cc ;
> 5 | var dd;;
    |       ^
  6 | var ee; ;
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Unnecessary semicolon (no-extra-semi) at __tests__/wpcs/fixtures/semicolons-fixture.js:5:8:
  3 | var bb
  4 | var cc ;
> 5 | var dd;;
    |        ^
  6 | var ee; ;
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Unnecessary semicolon (no-extra-semi) at __tests__/wpcs/fixtures/semicolons-fixture.js:6:9:
  4 | var cc ;
  5 | var dd;;
> 6 | var ee; ;
    |         ^
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Unnecessary escape character: \\\\\\" (no-useless-escape) at __tests__/wpcs/fixtures/strings-fixture.js:5:42:
  3 | var bb = 'Escape single quotes \\\\'within strings\\\\'';
  4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
> 5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
    |                                          ^
  6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Unnecessary escape character: \\\\\\" (no-useless-escape) at __tests__/wpcs/fixtures/strings-fixture.js:5:58:
  3 | var bb = 'Escape single quotes \\\\'within strings\\\\'';
  4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
> 5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
    |                                                          ^
  6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";
  7 | /* eslint-enable no-unused-vars */
  8 | 


error: Strings must use singlequote (quotes) at __tests__/wpcs/fixtures/strings-fixture.js:6:10:
  4 | var cc = 'Don\\\\'t escape double quotes \\"within strings\\"';
  5 | var dd = 'Needless escaped single quotes \\\\\\"within strings\\\\\\"';
> 6 | var ee = \\"Strings should be contained in single quotes, not double quotes\\";
    |          ^
  7 | /* eslint-enable no-unused-vars */
  8 | 


8 errors found.
6 errors potentially fixable with the \`--fix\` option."
`;
